### ISSUE TITLE
Setting up on a server with an existing world

### ISSUE THREAD

**Issue by JoeDobro93** [jsorrell/CarpetSkyAdditions#97]
Hello, I'm not new to Minecraft, but I'm coming back after many years away and not used to modding with Forge and am struggling to get a server running with this mod, which would allow me to keep generating void without needing to pregenerate any chunks and remove their blocks manually. I'll include a tldr at the end, but I figure more detail doesn't hurt when asking for tech help.

The background for this server is, I generated a world in vanilla in version 1.19.3, *attempted* to delete a 4000x4000 block square around spawn using Amulet, and then recreated the old Skyblock 2.1 map in the overworld and Nether. I set it up this way so that biomes and structure boundaries still exist but all blocks are gone. I found out, however, through building towards a jungle biome that only a small radius (closer to 600x600) of the world was actually deleted and natural terrain appeared. I tried other methods, such as pre-generating chunks and deleting the boxes, but there were just too many to delete which was way too slow, and some googling evenually led me to this page.

I tried installing Forge for the first time today, I'm trying to set this up on a server that I installed using it. I placed the Carpet mod and Carpet Sky Additions mod .jar files in the Forge server's /mod folder but it doesn't seem to be working. The world will load, but none of the Carpet commands appear in the console or if I op myself and try to use cheats. I did not install add the datapack since I didn't think I needed that, but if it's needed then let me know. The datapack seemed to be something to allow normally unobtainable items to become obtainable in skyblock, but I am looking for basically just vanilla skyblock so I didn't want these features.

When I try to load the server, I many warnings, many along the lines of "Mod file C:...fmlcore-1.19.3-44.1.20.jar is missing mods.toml file" and "Assets URL 'union:/C:..../.mcassestsroot' uses unexpected schema. The world does ultimately load, but it's just a regular vanilla world with none of the Carpet mod abilities. I tried creating the .config files with the default parameters that were shown in the other thread about using this on an existing world, but it didn't seem to have any effect. I also used an NBT editor to change the world generation from 'minecraft:noise' to 'carpetskyadditions:skyblock' and when I do this, the server won't even finish initializing.

**tldr; I would like to take an existing skyblock world from a vanilla server, and use this mod with it to generate void containing biomes and structure boundaries based on the original seed, but otherwise keep everything else entirely vanilla as far as game mechanics go. I tried many things that didn't work, so I could really use some help in getting this set up.**

Thanks in advance!

**Comment by jsorrell** [comment]
The problem is this is a Fabric mod, not a Forge mod. You need to use Fabric mod loader, not Forge.
You can find info here: https://fabricmc.net/wiki/start

You don't need the datapack or the config files if you don't want the added mod features, but the change to the NBT file is necessary.
But if you don't want end portals or ancient city portals generating, you can add `carpetskyadditions.conf` to your world folder containing:
```
generateEndPortals false
generateAncientCityPortals false
generateSilverfishSpawners false
```
